types/ext/dictionary.d.ts

```diff
@@ -210,7 +210,7 @@ export type TermDictionaryEntry = {
     /**
      * A list of inflections that was applied to get the term.
```

Comment:
Is this still the best description? It seems to not reflect the "hypothesis" concept. Also I wonder if "hypothesis" is the best word, as that is something I would expect more in some sort of logical deduction system. Here I wonder if it's just more `possibleInflections`, `potentialInflections`, `inflectionCandidates` or something like that?

Reply:
I was hoping someone would suggest something about that. What I'm going for is: 
- give the individual, atomic inflections a different name from the composite 
- not imply that the list of composite inflections is exhaustive

I've changed it to "inflectionPossibility" (the strings that make up an inflectionPossibility can be "inflections"), and updated the description. Better?

Reply:
Hmm, I think we need to think more carefully about this.

I think we have the following concepts:
* Inflection Rule: A single atomic operation that can be performed on a word to inflect it.
* Inflection Rule Chain: Multiple inflection rules applied in order. (I think it's not really a "possibility" at this point, that's more a function of how the layer above it uses this type.)
* Potential Inflection Rule Chains: A list of Inflection Rule Chains that are ways that a looked-up word can be reached from a dictionary headword.

And thinking from first principles about how this should be handled in code:
* "Inflection Rule" should almost certainly be a type (`InflectionRule`, or just `Rule` if we are in an appropriately named module), since it's very clearly a union of a couple different pieces of information. I would imagine the "source" should be on this type, since one could imagine building an Inflection Rule Chain from multiple Inflection Rules sourced from different places (although I guess your code doesn't currently do this).
* An "Inflection Rule Chain" could honestly just be a list, I'm not sure it really needs its own type (`InflectionRuleChain`), but I guess it would be okay as such to add a bit of semantic clarity and avoid `[][]` in the following bullet.
* "Potential Inflection Rule Chains" honestly feels like it should just be a variable (`potentialInflectionRuleChains`) pointing to the type `InflectionRuleChain[]` (or `Set<InflectionRuleChain>` since the order doesn't matter) at the location of whatever is using Inflection Rule Chains in a certain way.

Some things I considered while writing the above:
* I think "Inflection" alone is way too ambiguous whether it's referring to a process, a rule, a chain of roles, or the resulting word of after applying a rule. IMO it's best to have it refer to the overall process/concept, and be more explicit about calling things rules or results.
* I had to consider whether "deinflection" was more appropriate, but again, I think it's best that it refers to the process  (as opposed to rules, chains of rules, or a result of applying that chain). Also, since the same concepts would be used in inflection and deinflecting processes, I think the types can simply just refer to "inflection" since that is the more core concept (as opposed to the negated concept of deinflection). Then the module code and/or function names can of course refer to "deinflection" if that is what they are doing.
* We should be careful to support both agglutinative and fusional languages here. AFAIK, both of those use the concept of inflection, but fusional languages will just have shorter chains (usually of length 1 probably). I think that fits fine in the above breakdown. I suppose we also need to support analytic languages but I think that is not a problem as they will simply always be direct dictionary hits and avoid the deinflector entirely.
* We need to be careful to avoid the term "derivation" since that is a similar but separate concept in linguistics.

(This is just meant to be ground-up thinking as opposed to a comment on your PR. Feel free to take this and integrate it however you think it fits best. Also cc @toasted-nutbread)

Reply:
Lots of good points, thanks. I've kept `PossibleInflectionRuleChain` as a type for now because the source property is on it, and `InflectionRuleChain` as a string[]. Other than that, the nomenclature is now pretty much as you described it.

Reply:
I'm a bit late getting back to this, but overall I agree with the sentiment. I've had some thoughts about the deinflector class and its types being kinda sloppy in general.

The only place I differ is just in naming preference, in that I like `InflectionRuleChainCandidates` over `PossibleInflectionRuleChains` because it keeps the same suffix as the other types and "candidate" sounds more like a programming term (e.g. "release candidate").

---

The only other thing that comes to my mind with regard to the term "inflection" is "what about things that aren't strictly-speaking an inflection?" For example, text transformations can still occur in other ways, and I think I've probably mentioned that the extension 10ten does this for dates. I think I may have touched on this briefly in https://github.com/FooSoft/yomichan/issues/1410 a long time ago.

Overall, this last part of my comment goes beyond the scope of this change, but I wonder if perhaps eventually the idea of an "Inflection" should be further generalized to just a "Transformation" or something like that, and then potentially categorize it as "inflection" or "`<insert-something-else-here>`".

Reply:
> I've kept PossibleInflectionRuleChain as a type for now because the source property is on it, and InflectionRuleChain as a string[]
@StefanVukovic99 Ok -- I think it's fine for now, but in the end the source should probably be on an Inflection Rule.

toasted-nutbread ideas all sound good so I think we can maybe do a further generalization in the future.